authors/apps/articles/views.py

```diff
@@ -699,3 +701,69 @@ def get(self, request):
             "bookmarks": bookmarked_articles
         }
         return Response(data=bookmarks, status=status.HTTP_200_OK)
+
+
+class GetAllArticleReports(generics.ListAPIView):
+    """ admin class for getting all article reports """
+    serializer_class = ReportSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ReportJSONRenderer,)
+
+    def get(self, request):
+        """ get endpoint for fetching actual reports """
+        user = request.user
+        if user.is_staff is False:
```

Comment:
@BrianSerem, you can check out a better implementation of this[ here](https://www.django-rest-framework.org/api-guide/permissions/#isadminuser)